rclcpp/src/rclcpp/parameter_service.cpp

```diff
@@ -17,11 +17,78 @@
 #include <algorithm>
 #include <vector>
 
+#include "rcl/rcl.h"
+
+#include <string.h>
+
 using rclcpp::parameter_service::ParameterService;
 
 ParameterService::ParameterService(const rclcpp::node::Node::SharedPtr node)
 : node_(node)
 {
+  int argc;
+  char** argv;
+  argc = rcl_get_argv(&argv);
+  for (int i = 1; i < argc; ++i)
+  {
+    char * arg = strdup(argv[i]);
+    if (arg == NULL) {
+      break;
+    }
+    int len = strlen(arg);
+    // Minimum expression is _n:=v
```

Comment:
sounds good to me. I think the general guideline is that all the logic that is neither rmw specific nor language specific should land in `rcl`. For now there is still a lot in the `rclcpp` layer because `rclpy` was/is lagging far behind and it was faster to prototype in c++ than in c